README.md

```diff
@@ -108,6 +108,18 @@ such a server can be launched with `python -m smtpd -n -c DebuggingServer localh
 
 Alternatively only unit tests can be run by doing `cargo test --lib`.
 
+## Troubleshooting
+
+These are general steps to be followed when troubleshooting SMTP related issues.
+
+- Ensure basic connectivity, ensure requisite ports are open and daemons are listening.
+- Confirm that your service provider allows traffic on the ports being used for mail transfer. (traffic on ports 25, 2525, 587, and 465 is commonly blocked by some service providers.)
+- Check SMTP relay authentication and configuration.
+- Validate your DNS records. (DMARC, SPF, DKIM, MX)
+- Verify your SSL/TLS certificates are setup properly. (ensure cert and bundled ca-certificates are up to date and valid, check file permissions)
+- Determine if SSL and/or STARTTLS are supported and setup as required by remote hosts.
```

Comment:
That shouldn't matter on port 25 at least. TLS is not meant to be enforced here like it is on 587 (optional via STARTTLS, but can be configured as required) and 465.

Reply:
I guess removing line 120 would suffice to keep things simple, but I would recommend keeping line 119 otherwise you may wind up with uncertain users reporting issues more likely to be with rust-tls and/or local certificate stores.

Reply:
> otherwise you may wind up with uncertain users reporting issues

Where I'm a maintainer on a project, I generally try not to exhaustively worry about if certain documentation is missing like troubleshooting due to that reason :)

When a pattern of users reporting an issue emerges, it's a good indicator at that point to add to the docs :sweat_smile: 

For example with line 119, if  you keep it and still get issues related to it, you might find the issues are often resolved in the same way, advising how to verify the certificate, in which case you'd revise line 119 then to include that knowing it's a common enough struggle users run into.

Removing line 120 is fine for that same reason, you can let the users reports refine the contents from the already good base document your PR covers :+1:

Reply:
Understood.

I suppose removing the parenthetical on line 119 would again be good. Allowing for users to interpret any troubleshooting involving SSL certs as they see fit and then fleshing out the more common issues over time makes perfect sense.